File: app/src/main/java/io/toru/retrofitandmvp/network/StackOverFlowApi.java
package io.toru.retrofitandmvp.network;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by toru on 2016. 8. 15..
 */
public interface StackOverFlowApi {
    @GET("/2.2/questions?order=desc&sort=creation&site=stackoverflow")
    Call<StackOverFlowQuestion> loadQuestions(@Query("tagged") String tags);
}

File: kotlin-app/src/main/kotlin/io/toru/retrofitandmvp/kotlin/repository/StackOverFlowApi.kt
package io.toru.retrofitandmvp.kotlin.repository

import io.toru.retrofitandmvp.kotlin.repository.model.StackOverFlowQuestion
import retrofit2.Retrofit
import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable
import javax.inject.Inject


class StackOverFlowApi @Inject constructor(private val retrofit: Retrofit) {

    fun loadQuestions(@Query("tagged") tags: String): Observable<StackOverFlowQuestion> {
        return retrofit.create(Api::class.java).loadQuestions(tags)
    }

    fun loadQuestioniOS(@Query("tagged") tags: String): Observable<StackOverFlowQuestion> {
        return retrofit.create(Api::class.java).loadQuestioniOS(tags)
    }

    interface Api {
        @GET("/2.2/questions?order=desc&sort=creation&site=stackoverflow")
        fun loadQuestions(@Query("tagged") tags: String): Observable<StackOverFlowQuestion>

        @GET("/2.2/questions?order=desc&sort=creation&site=stackoverflow")
        fun loadQuestioniOS(@Query("tagged") tags: String): Observable<StackOverFlowQuestion>
    }
}